ddtrace/sampler.py

```diff
@@ -199,6 +210,27 @@ def __init__(
             )
             self.default_sampler = SamplingRule(sample_rate=default_sample_rate)
 
+    def _parse_rules_from_env_variable(self, rules):
+        sampling_rules = []
+        if rules is not None:
+            json_rules = []
+            json_rules = json.loads(rules)
+            for rule in json_rules:
+                if "sample_rate" not in rule:
+                    raise KeyError("No sample_rate provided for the following rule {%s}", rule)
+                sample_rate = float(rule["sample_rate"])
+                service = rule.get("service", SamplingRule.NO_RULE)
+                name = rule.get("name", SamplingRule.NO_RULE)
+                # sampling_rule = SamplingRule(sample_rate=sample_rate, service=service, name=name)
+                try:
+                    sampling_rule = SamplingRule(sample_rate=sample_rate, service=service, name=name)
+                except ValueError:
+                    raise ValueError(
+                        "SamplingRule() must be greater than or equal to 0.0 and less than or equal to 1.0", rule
+                    )
```

Comment:
If we add more validation to `SamplingRule` in the future then this error message might not make sense anymore.

It might be better to do something like this:

```suggestion
                except ValueError as e:
                    raise ValueError("Error creating sampling rule {%s}: {%s}".format(rule, e))
```


Basically just pass through whatever the original error message is, but add the context of "for this specific rule" to add additional context needed.

What do you think?

Reply:
I like this approach 👍  makes it more flexible for future error handling as you pointed out.